compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1405,11 +1405,16 @@ public void visit(BLangRecordVarRef varRefExpr) {
         varRefExpr.symbol = new BVarSymbol(0, recordSymbol.name,
                 env.enclPkg.symbol.pkgID, bRecordType, env.scope.owner);
 
-        if (varRefExpr.restParam == null) {
+        if (restParam == null) {
             bRecordType.sealed = true;
             bRecordType.restFieldType = symTable.noType;
-        } else {
+        } else if (restParam.type == symTable.semanticError) {
             bRecordType.restFieldType = symTable.mapType;
+        } else {
+            // Since the variable ref for a map returns the Map type and the model stores the constraint of the map type
```

Comment:
Shall we update the comment to something like:
Rest variable type of Record ref (record destructuring assignment) is a map<T> where T is the broad type of all fields that are not specified in the destructuring pattern. Here we set the rest type of record type to T.

Please feel free to change the message appropriately

Reply:
Fixed with 60bfc55